src/spec.ts

```diff
@@ -336,6 +361,9 @@ function fieldDefIndex(spec: GenericSpec<GenericUnitSpec<any, any>>, dict: any =
   } else if (isFacetSpec(spec)) {
     accumulate(dict, vlEncoding.fieldDefs(spec.facet));
     fieldDefIndex(spec.spec, dict);
+  } else if (isRepeatSpec(spec)) {
+    accumulate(dict, vlEncoding.fieldDefs(spec.spec));
+    fieldDefIndex(spec.spec, dict);
```

Comment:
Well, this is only used in tests. So we can assume that tests don't sue repeat for now.